### DISCUSSION TITLE
How to add path prefix to links?

### DISCUSSION THREAD

**Discussion by Geoc2022** [community/community#43451] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have moved files from my profile github pages site (username.github.io) to a repository github pages site (username.github.io/reponame). The site files contain absolute links (e.g., "/site.js") that have broken since they link to the original site ("username.github.io/site.js") rather than the new site ("username.github.io/reponame/site.js").

Is it possible to specify a "path prefix," "reponame" in this case, for links in github pages such that all links in username.github.io/reponame have a prefix such that they direct inside the site — similar to how [Gatsby interacts with Github pages](https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/how-gatsby-works-with-github-pages/#deploying-to-a-path-with-pathprefix)?

Although a simple solution is to replace the links, is there a solution for cases with many links?

**Comment by antoninpicard** [comment, marked as answer]
Dans ton repos GitHub, accédez au fichier de configuration de ton site de pages (normalement _config.yml).
Ajoutez la ligne suivante au fichier de configuration : baseurl: /reponame. Cela indiquera à Jekyll (le moteur de génération de sites utilisé par GitHub Pages) de préfixer tous les liens du site avec /reponame.
Modifiez tous les liens du site en utilisant un chemin relatif au lieu d'un chemin absolu. Par exemple, au lieu d'utiliser /site.js, utilisez site.js. Cela garantira que les liens fonctionnent correctement même lorsque le préfixe /reponame est ajouté.
Il existe également des plugins Jekyll qui peuvent t'aider à gérer cela de manière plus automatique, mais le processus ci-dessus devrait fonctionner pour la plupart des cas d'utilisation.
